request_handling.py

```diff
@@ -0,0 +1,42 @@
+import urllib.request as urllib2
+import requests
+from bs4 import BeautifulSoup
+from neon_utils.skills.neon_skill import LOG
+
+
+class RequestHandler:
+
+    def __init__(self):
+        self.found_shops = []
+
+    def existing_lang_check(self, user_lang, url):
+        link = url+user_lang+'/directory/'
+        response = requests.get(link)
+        if response.status_code == 200:
+            LOG.info('This language is supported')
+            return True
+        else:
+            LOG.info('This language is not supported')
+            return False
+
+    def get_shop_data(self, url, user_request):
+        page = urllib2.urlopen(url)
```

Comment:
Consider skipping parsing if `self.found_shops` is already populated; later this could be a dict indexed by url

Reply:
Do you mean to skip it if there is already some info in found_shops? Or check firstly the existence of requested shop there?

Reply:
I think the directories are fairly static, so if it was downloaded once we can probably skip it.. But that's a good point that maybe we should check again if the requested store isn't found.. Mostly I was thinking that parsing the web page can add some latency to responses